packages/host/app/components/operator-mode/submode-layout.gts

```diff
@@ -191,6 +199,76 @@ export default class SubmodeLayout extends Component<Signature> {
     this.suppressSearchClose = false;
   });
 
+  @cached
+  private get roomResources() {
+    let resources = new TrackedMap<string, RoomResource>();
+    for (let roomId of this.matrixService.rooms.keys()) {
+      resources.set(
+        roomId,
+        getRoom(this, () => roomId),
+      );
+    }
+    return resources;
+  }
+
+  @use private findUnseenMessage = resource(() => {
```

Comment:
Maybe I'm oversimplifying but can't `unseenMessage` just be a getter that would read the resolved room's (last) messages and check if it wasn't read yet?

Reply:
I need this https://github.com/cardstack/boxel/pull/1378/files#diff-9c8b5924f28c9a8b13c32f91f616aef178ea7ce8bd3360a7148917ba29cd2d0bR247-R252 to make the toast disappear after three seconds. I don't know if that is possible to do using getter.